src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunRestartTest.java

```diff
@@ -175,4 +184,96 @@ public DescriptorImpl() {
         }
     }
 
+    @Issue("JENKINS-43055")
+    @Test
+    public void flowExecutionListener() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("echo 'Running for listener'; sleep 0; semaphore 'wait'; sleep 0; error 'fail'", true));
+                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("wait/1", b);
+                ExecListener listener = ExtensionList.lookup(FlowExecutionListener.class).get(ExecListener.class);
+                assertNotNull(listener);
+                assertEquals(1, listener.started);
+                assertEquals(0, listener.resumed);
+                assertEquals(0, listener.finished);
+            }
+        });
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                WorkflowRun b = p.getLastBuild();
+                assertTrue(b.isBuilding());
+                SemaphoreStep.success("wait/1", null);
+
+                story.j.assertBuildStatus(Result.FAILURE, story.j.waitForCompletion(b));
+                story.j.assertLogContains("Running for listener", b);
+
+                ExecListener listener = ExtensionList.lookup(FlowExecutionListener.class).get(ExecListener.class);
+                assertNotNull(listener);
+                assertEquals(1, listener.started);
```

Comment:
Potentially confusing to readers. This `started` is not the same as the one in the first step: `onRunning(…, true)` increments _both_ fields, but we are using a new listener!

This is why I wonder whether having a single method overloaded to two events is wise.